dspace-api/src/main/java/org/dspace/service/impl/ClientInfoServiceImpl.java

```diff
@@ -119,9 +183,8 @@ private String getXForwardedForIpValue(String remoteIp, String xForwardedForValu
                not equal to the proxy
             */
             if (!StringUtils.equals(remoteIp, xfip) && StringUtils.isNotBlank(xfip)
-                    //if we have trusted proxies, we'll assume that they are not the client IP
-                    && (trustedProxies == null || !isRequestFromTrustedProxy(xfip))) {
-
+                    // if we have trusted proxies, we'll assume that they are not the client IP
```

Comment:
I've detected a small bug here.
xForwardedForValue can contain `ip1, ip2`
This will check `ip1` first, and ` ip2` next. If `ip2` is trusted, it will return it anyway due to the space
So this is missing a trim prior to the if

Reply:
Good catch, that would've been a very difficult one to locate.  I've fixed it in https://github.com/DSpace/DSpace/pull/3171/commits/c9fcacfd83343d5e874343940acb9526b678a262